packages/core/core.ts

```diff
@@ -90,6 +94,50 @@ export async function startFuzzingNoInit(
 	return Promise.resolve().then(() => fuzzerFn(fuzzFn, fuzzerOptions));
 }
 
+function prepareLibFuzzerArg0(fuzzerOptions: string[]): string {
+	// When we run in a libFuzzer mode that spawns subprocesses, we create wrapper script
+	// that can be used as libFuzzer's argv[0]. In the fork mode, the main libFuzzer process
+	// uses argv[0] to spawn further processes that perform the actual fuzzing.
+	const libFuzzerSpawnsProcess = fuzzerOptions.some(
+		(flag) =>
+			flag.startsWith("-fork=") ||
+			flag.startsWith("-jobs=") ||
+			flag.startsWith("-merge=")
+	);
+
+	// Return a fake argv[0] to start the fuzzer if libFuzzer does not spawn new processes.
+	if (!libFuzzerSpawnsProcess) {
+		return "unused_arg0_report_a_bug_if_you_see_this";
+	}
+
+	const jazzerArgs = process.argv.filter(
+		(arg) => arg !== "--" && fuzzerOptions.indexOf(arg) === -1
+	);
+
+	const isWindows = process.platform === "win32";
+
+	const newLine = isWindows ? "\r\n" : "\n";
+	const scriptContent =
+		(isWindows ? "@echo off" : "#!/usr/bin/env sh") +
+		newLine +
+		`cd "${process.cwd()}"` +
+		newLine +
+		jazzerArgs.map((s) => '"' + s + '"').join(" ") +
+		" -- " +
+		(isWindows ? "%*" : "$@") +
+		newLine;
+
+	const scriptTempFile = tmp.fileSync({
```

Comment:
Should we extract the template and file generation out into its own function or even file as it's not directly related to the fuzzer parameter handling?

Reply:
Extracting the script generation into a dedicated function makes the code cleaner. Keeping these functions close to each other in the same file makes the code easier to follow since they are highly related. WDYT?

Reply:
Looks good, thx.